File: src/targets/ruby/native/fixtures/headers.rb
require 'uri'
require 'net/http'

url = URI("https://httpbin.org/headers")

http = Net::HTTP.new(url.host, url.port)
http.use_ssl = true

request = Net::HTTP::Get.new(url)
request["accept"] = 'application/json'
request["x-foo"] = 'Bar'
request["x-bar"] = 'Foo'
request["quoted-value"] = '"quoted" \'string\''

response = http.request(request)
puts response.read_body

File: src/targets/c/libcurl/fixtures/headers.c
CURL *hnd = curl_easy_init();

curl_easy_setopt(hnd, CURLOPT_CUSTOMREQUEST, "GET");
curl_easy_setopt(hnd, CURLOPT_WRITEDATA, stdout);
curl_easy_setopt(hnd, CURLOPT_URL, "https://httpbin.org/headers");

struct curl_slist *headers = NULL;
headers = curl_slist_append(headers, "accept: application/json");
headers = curl_slist_append(headers, "x-foo: Bar");
headers = curl_slist_append(headers, "x-bar: Foo");
headers = curl_slist_append(headers, "quoted-value: \"quoted\" 'string'");
curl_easy_setopt(hnd, CURLOPT_HTTPHEADER, headers);

CURLcode ret = curl_easy_perform(hnd);